CASM-3422: Security: Container Image signature verification (#5362)
Cray-HPE/docs-csm: operations/kubernetes/Kyverno.md

"cray-ceph-csi-cephfs-nodeplugin","DaemonSet","ceph-cephfs","disallow-privileged-containers","validation error: Privileged mode is disallowed. The fields spec.containers[*].securityContext.privileged and spec.initContainers[*].securityContext.privileged must be unset or set to `false`. Rule autogen-privileged-containers failed at path /spec/template/spec/containers/0/securityContext/privileged/"
```

## What is new in the HPE CSM 1.6 release and above

1. Kyverno is upgraded from 1.9.5 version to 1.10.7 version and is now available for customers as part of the HPE CSM 1.6 release.

>    This is a major upgrade with many new features and bug fixes. For complete list please refer to the link [CHANGELOG](https://github.com/kyverno/kyverno/blob/main/CHANGELOG.md)

Reviewer: ```suggestion
   This is a major upgrade with many new features and bug fixes. For a complete list, refer to the [Kyverno CHANGELOG](https://github.com/kyverno/kyverno/blob/main/CHANGELOG.md).
```

Reviewer 2: *Kyverno

2. Container image signing and verification using Kyverno policy.

### Container image signing and verification using Kyverno policy

Container images are signed and verified using a Kyverno policy for software supply chain security. For more information, refer to the link

> [Verify image signatures](https://kyverno.io/docs/writing-policies/verify-images)

Reviewer: ```suggestion
[Verify image signatures](https://kyverno.io/docs/writing-policies/verify-images).
```

      generation: 3
      labels:
        app.kubernetes.io/managed-by: Helm
      name: check-image
      resourceVersion: "97179892"
      uid: 3a65f533-1dc8-472b-bcdb-93f264a04f21
    spec:
      background: true
      failurePolicy: Fail
      rules:
      # The below section provides exceptions for docker-kubectl* named resource in cosign-test namespace
      - exclude:
          any:
          - resources:
              names:
              - docker-kubectl*
              namespaces:
              - cosign-test
        match:
          any:
          - resources:
              kinds:
              - Pod
              namespaces:
              - '*'
        name: check-image
        verifyImages:
        - attestors:
          - count: 1
            entries:
            - keys:
                publicKeys: |-
                  {{- range $name, $key := $keys }}
                  {{ $key | b64dec | trim | indent 14 }}
                  {{- end }}
          imageReferences:
          - '*'
          mutateDigest: false
          required: true
          verifyDigest: false
      validationFailureAction: Audit
      webhookTimeoutSeconds: 30
    ```

>     The unsigned container images added as exceptions won't be reported as policy violations in the policy report. To understand more about adding exceptions, refer to this link

Reviewer: ```suggestion
    The unsigned container images added as exceptions will not be reported as policy violations in the policy report. To understand more about adding exceptions, refer to this link
```

>     [Adding Exceptions](https://release-1-9-0.kyverno.io/docs/writing-policies/match-exclude/#match-statements)

Reviewer: ```suggestion
    [Adding Exceptions](https://release-1-9-0.kyverno.io/docs/writing-policies/match-exclude/#match-statements).
```

      resourceVersion: "97174145"
      uid: 3a65f533-1dc8-472b-bcdb-93f264a04f21
    spec:
      background: true
      failurePolicy: Fail
      rules:
      - exclude:
          any:
          - resources:
              names:
              - sample-name
              namespaces:
              - sample-ns
        match:
          any:
          - resources:
              kinds:
              - Pod
              namespaces:
              - '*'
        name: check-image
        verifyImages:
        - attestors:
          - count: 1
            entries:
            - keys:
                publicKeys: |-
                  {{- range $name, $key := $keys }}
                  {{ $key | b64dec | trim | indent 14 }}
                  {{- end }}
            # This section adds a public key stored in the secret customer-test and in cosign-test namespace
            - keys:
                publicKeys: k8s://cosign-test/customer-secret
                signatureAlgorithm: sha256
          imageReferences:
          - '*'
          mutateDigest: false
          required: true
          verifyDigest: false
      validationFailureAction: Audit
      webhookTimeoutSeconds: 30
    ```

>     The container images succesfully signed by the Customers using their own private key, won't be reported as policy violations in the policy report.

Reviewer: ```suggestion
    The container images successfully signed by the Customers using their own private key, will not be reported as policy violations in the policy report.
```
